CesiumGS/cesium
# CreditDisplay.addCreditToNextFrame and CreditDisplay.addStaticCredit

This fixes https://github.com/CesiumGS/cesium/issues/6215 in preparation for adding credits to lightbox for use with the geocoder.

Given the confusion discussed in the issue, I deprecated existing functions in favor of `addCreditNextFrame` and `addStaticCredit`/`removeStaticCredit`, the latter of which now takes  `showOnScreen` into account.

This also
- adds a bit more documentation showing usage examples, and notes that `addCreditNextFrame` is mostly used in internal cases
- adds accessors Viewer.creditDisplay and CesiumWidget.creditDisplay for easier access

To test:

1.
`viewer.creditDisplay.addStaticCredit(new Cesium.Credit("My credit"));`
should show "My credit" in the lightbox

2.
`viewer.creditDisplay.addStaticCredit(new Cesium.Credit("My onscreen credit", true));`
should show "My onscreen credit" onscreen

`addCreditToNextFrame` is called internally, so

3. Confirm credits from imagery are added. For example the following is added to the lightbox for Bing Maps:
```
 © 2023 Microsoft Corporation
  Earthstar Geographics SIO
  <LOGO>
 ```

4. Load the ArcGIS MapServer example. There should be a default token credit warning on screen.
